lib/THC/generic/THCStorage.cu

```diff
@@ -61,7 +61,21 @@ void THCStorage_(resize)(THCState *state, THCStorage *self, ptrdiff_t size)
                                  size * sizeof(real),
                                  THCState_getCurrentStream(state));
     if(err != cudaSuccess) {
-      THCHeapUpdate(state, -size * sizeof(real));
+      if(self->flag & TH_STORAGE_FREEMEM) {
+         // Try reallocating memory on failure
+         THCudaCheck(
+           (*self->allocator->free)(self->allocatorContext, self->data));
+         THCHeapUpdate(state, -self->size * sizeof(real));
+         self->data = NULL;
```

Comment:
I think it would be better to reallocate the previous size if this bigger allocation still fails. The docs say that after resize tensor content is unspecified so it's ok, but having NULL data pointer seems dangerous.